androidApp/src/main/java/com/mbta/tid/mbta_app/android/util/SocketMessage.kt

```diff
@@ -1,42 +1,25 @@
 package com.mbta.tid.mbta_app.android.util
 
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonPrimitive
 import org.phoenixframework.Message
 
 fun decodeMessage(rawMessage: String): Message {
-    // https://github.com/dsrees/JavaPhoenixClient/blob/1.3.1/src/main/kotlin/org/phoenixframework/Defaults.kt#L68
-    val parseValue: (String) -> String? = { value ->
-        when (value) {
-            "null" -> null
-            else -> value.replace("\"", "")
-        }
-    }
+    val message = Json.parseToJsonElement(rawMessage).jsonArray
+    val joinRef = message[0].jsonPrimitive.content
```

Comment:
Per [the `JsonPrimitive.content` docs](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-primitive/content.html), `content` will be `"null"` if the value is `null`, and [`contentOrNull`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/content-or-null.html) is the correct way to get `null` from `null`. (The Kotlin serialization library stores JSON internally in a weird half-hydrated state that is optimized specifically for serialization and deserialization more than for in-memory processing, which I learned way too much about as part of [FeatureProperties](https://github.com/mbta/mobile_app/blob/78e046e2de357c0d7406701054ed51d0163e758a/shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/map/style/FeatureProperties.kt#L5-L11).)